packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java

```diff
@@ -1,24 +1,57 @@
 package software.uncharted.terarium.hmiserver.models.dataservice;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
 import software.uncharted.terarium.hmiserver.annotations.TSModel;
 
 @RequiredArgsConstructor
 @TSModel
+@Slf4j
 public enum AssetType {
-	WORKFLOW("workflow"),
-	MODEL("model"),
-	DATASET("dataset"),
-	SIMULATION("simulation"),
-	DOCUMENT("document"),
-	CODE("code"),
-	MODEL_CONFIGURATION("model-configuration"),
-	ARTIFACT("artifact"),
-	PUBLICATION("publication"),
-	NOTEBOOK_SESSION("notebook-session"),
+	@JsonProperty("workflow")
+	WORKFLOW,
+
+	@JsonProperty("model")
+	MODEL,
+
+	@JsonProperty("dataset")
+	DATASET,
+
+	@JsonProperty("simulation")
+	SIMULATION,
+
+	@JsonProperty("document")
+	DOCUMENT,
+
+	@JsonProperty("code")
+	CODE,
+
+	@JsonProperty("model-configuration")
+	MODEL_CONFIGURATION,
+
+	@JsonProperty("artifact")
+	ARTIFACT,
+
+	@JsonProperty("publication")
+	PUBLICATION,
+
+	@JsonProperty("notebook-session")
+	NOTEBOOK_SESSION,
 	;
 
-	@JsonValue
-	private final String value;
+	public static AssetType getAssetType(final String assetTypeName, final ObjectMapper objectMapper) throws ResponseStatusException {
```

Comment:
Why do we need to pass `objectMapper`, can we not instantiate one in this method?

Reply:
Enums cannot Autowire/inject beans.  The main workarounds for this is to have a factory class that provides the enum and can do the heavy lifting.  Will that be better?

Reply:
No, all good, thanks